dev/date-picker.html

```diff
@@ -9,6 +9,17 @@
     <script type="module">
       import '@vaadin/date-picker';
       import '@vaadin/tooltip';
+      const isDateDisabled = (date) => {
+        // Exclude weekends and the 16th day of each month:
+        const checkDate = new Date(0, 0);
+        checkDate.setFullYear(date.year);
+        checkDate.setMonth(date.month);
+        checkDate.setDate(date.day);
+        return checkDate.getDay() === 0 || checkDate.getDay() === 6 || checkDate.getDate() === 16;
+      }
```

Comment:
We recently had a brief internal discussion regarding the parameter format for the 
`isDateDisabled` function, considering the diverse range of use cases. Two potential options:

Option 1: `DatePickerDate` (`{year: 2023, month: 10, day: 10}`):
- More structured and explicit, allowing users to access individual components of the date easily.
- Might be easier for disabling specific weekdays or days of months

Option 2: ISO Formatted Date String (`"2023-11-10"`):
- Familiar format, commonly used in various contexts.
- Same format as the component's `value`
- Might be easier for disabling specific dates (stored in DB as local date for example)

If a decision between these two options proves challenging, we could consider offering flexibility by accepting both ISO formatted date strings and date objects with individual components. For example, an extended `DatePickerDate` could include both formats as follows:

```
{
  year: 2023,
  month: 10,
  day: 10,
  date: "2023-11-10"
}
```

@rolfsmeds @yuriy-fix

Reply:
Would the ISO date be optional or should we expect that the `DatePickerDate` structure always contains all four fields?

Reply:
Modifying `DatePickerDate` might be breaking, so better introduce a new type or extend it.

Reply:
For now it seems to be working well enough with `DatePickerDate`. If a developer needs more detailed logic it's easy enough to convert it to a Javascript Date object.